assets/js/modules/analytics-4/datastore/properties.js

```diff
@@ -99,6 +101,38 @@ const baseActions = {
 			return { response, error };
 		}() );
 	},
+
+	/**
+	 * Sets the given property and related fields in the store.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @param {string} propertyID GA4 property ID.
+	 * @return {Object} A Generator function.
+	 */
+	selectProperty( propertyID ) {
+		invariant( isValidPropertySelection( propertyID ), 'A valid propertyID selection is required.' );
+
+		return ( function* () {
+			const registry = yield Data.commonActions.getRegistry();
+
+			registry.dispatch( STORE_NAME ).setPropertyID( propertyID );
+			registry.dispatch( STORE_NAME ).setWebDataStreamID( '' );
+			registry.dispatch( STORE_NAME ).setMeasurementID( '' );
+
+			if ( PROPERTY_CREATE === propertyID ) {
+				return;
+			}
+
+			yield webDataStreamActions.waitForWebDataStreams( propertyID );
+
+			const webdatastream = registry.select( STORE_NAME ).getMatchingWebDataStream( propertyID );
+			if ( webdatastream ) {
+				registry.dispatch( STORE_NAME ).setWebDataStreamID( webdatastream.name.split( '/' ).pop() );
```

Comment:
We should probably extract a utility for this in the future, but this is okay for now.

Reply:
@aaemnnosttv it will be done in another ticket.